pkg/apis/httpEndpoint/v1alpha1/types.go

```diff
@@ -51,11 +53,43 @@ func (h HTTPEndpoint) NameValuePairs() []common.NameValuePair {
 	return h.Spec.Headers
 }
 
+// nil returns a bool indicating if a tls private key has a secret reference
+func (h HTTPEndpoint) HasTLSPrivateKeySecret() bool {
+	return h.Spec.ClientTLS != nil && h.Spec.ClientTLS.PrivateKey != nil && h.Spec.ClientTLS.PrivateKey.SecretKeyRef != nil && h.Spec.ClientTLS.PrivateKey.SecretKeyRef.Name != ""
```

Comment:
Please use the `GetXxx` methods, which have built-in nil-checks, to make this more readable. Something like:


```suggestion
	return h.Spec.GetClientTLS().GetPrivateKey().GetSecretKeyRef().GetName() != ""
```

Reply:
Same for the methods below

Reply:
These methods do not exist on the spec.

Reply:
Duh, I was thinking of protobuf where they are auto-generated.

Still, thoughts on adding them? Something like

```go
func (h HTTPEndpoint) GetTLSPrivateKey() *common.TLSDocument {
	if h.Spec.ClientTLS != nil && h.Spec.ClientTLS.PrivateKey != nil {
		return h.Spec.ClientTLS.PrivateKey
	}
	return nil
}

// HasTLSPrivateKeySecret returns a bool indicating if a tls private key has a secret reference
func (h HTTPEndpoint) HasTLSPrivateKeySecret() bool {
	privKey := h.GetTLSPrivateKey()
	return privKey != nil && privKey.SecretKeyRef != nil && privKey.SecretKeyRef.Name != ""
}
```

Reply:
TBH I don't like adding those as the code that would use them (the HasX methods) isn't likely to change much, and its usage will almost always come from outside the package where this change wouldn't affect readability much..